### ISSUE TITLE
Add state keys to path result

### ISSUE THREAD

**Issue by nreinicke** [NREL/routee-compass#240]
Looking at an example path returned from compass:

```
{'features': [{'geometry': {'coordinates': [[-104.96916961669922,
      39.77902603149414],
     [-104.96910095214844, 39.77896499633789],
     [-104.96887969970703, 39.77878189086914],
     [-104.96884155273438, 39.778751373291016]],
    'type': 'LineString'},
   'id': 11305,
   'properties': {'access_cost': 0.0,
    'edge_id': 11305,
    'result_state': [0.00137196828747872,
     0.055087682159255455,
     0.025957737726286807],
    'traversal_cost': 0.0181789351125543},
   'type': 'Feature'},
 ...
}
```

~It's not immediately clear what value is what in `result_state`. While we can cross reference this information from `result.route.state_model`,~

It's impossible to tell which value is which in `result_state` since the indices are not captured directly in the result.

 it might be more ergonomic to have them in the path directly, maybe something like `result_state_{state_feature}_{state_feature_unit}`.

**Comment by robfitzgerald** [comment]
right. this was supposed to be solved by the "state" section of the "route" object:

```json
            "state_model": {
                "battery_state": {
                    "format": {
                        "floating_point": {
                            "initial": 100.0
                        }
                    },
                    "type": "soc",
                    "unit": "percent"
                },
                "distance": {
                    "distance_unit": "miles",
                    "initial": 0.0
                },
                "energy_electric": {
                    "energy_unit": "kilowatt_hours",
                    "initial": 0.0
                },
                "time": {
                    "initial": 0.0,
                    "time_unit": "minutes"
                }
            }
```
but i'm realizing the index is missing from each object. or, since it's a map, we can't guarantee ordering. so, _i think_ that means this task is either to add the index:

```json
                "battery_state": {
                    "index": 0,
                    "format": {
                        "floating_point": {
                            "initial": 100.0
                        }
                    },
                    "type": "soc",
                    "unit": "percent"
                }
```

or make these an array:

```json
            "state_model": [
                {
                    "name": "battery_state",
                    "format": {
                        "floating_point": {
                            "initial": 100.0
                        }
                    },
                    "type": "soc",
                    "unit": "percent"
                }
            ]
```

thoughts?

**Comment by nreinicke** [comment]
Yeah those are both good solutions. I think I would favor adding the index into the entry specifically since it makes it obvious what it means.

**Comment by robfitzgerald** [comment]
Addressed via #251.
